<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>
  <arg name="Kp" default="0.46"/>
  <arg name="Ki" default="0.008"/>
  <arg name="Kd" default="0.2" />
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->

  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="camera_name" default="/raspicam_node" />
  <arg name="camera_frame" default="camera" />
  <arg name="image_topic" default="image" />

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />


  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml" ns="$(arg node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>

  <node pkg="assignment6_trackingandfollowing" type="april_tag_tracking.py" name="april_tag_tracking">
    <param name="Kp" value="$(arg Kp)"/>
    <param name="Ki" value="$(arg Ki)"/>
    <param name="Kd" value="$(arg Kd)" />
  </node>

</launch>
